main/inc/lib/main_api.lib.php

```diff
@@ -43,6 +43,7 @@
 define('COURSE_STUDENT', 14);   //student subscribed in a course
 define('SESSION_STUDENT', 15);  //student subscribed in a session course
 define('COURSE_TUTOR', 16); // student is tutor of a course (NOT in session)
+define('ROLE_INVITED', 20);
```

Comment:
INVITEE is a new role of a user that is invited to a (non-public) course and can use it but does not generate tracking information (it's a ghost, we could call him GHOST but it would be even weirder). It is a role used in Moodle and apparently some people are very attached to it. Although we don't really like adding a role just for that, it is true that there is currently no other way to get to the same effect. Check BT#9070 for details, but my explanation here sums it up.